dali/core/mm/default_resources.cc

```diff
@@ -163,6 +159,50 @@ bool UseVMM() {
   return value;
 }
 
+size_t MallocThreshold() {
+  char *env = getenv("DALI_MALLOC_POOL_THRESHOLD");
+  int len = 0;
+  if (env && (len = strlen(env))) {
+    for (int i = 0; i < len; i++) {
+      bool valid = std::isdigit(env[i]) || (i == len - 1 && (env[i] == 'k' || env[i] == 'M'));
+      if (!valid) {
+        DALI_FAIL(make_string(
+          "DALI_MALLOC_POOL_THRESHOLD must be a number, optionally followed by 'k' or 'M', got: ",
+          env));
+      }
+    }
+    size_t s = atoll(env);
+    if (env[len-1] == 'k')
+      s <<= 10;
+    else if (env[len-1] == 'M')
+      s <<= 20;
+    return s;
+  } else {
+    char *mmap = getenv("MALLOC_MMAP_MAX_");
+    if (mmap && !atoi(mmap))
+      return 0;
+    char *thresh = getenv("MALLOC_MMAP_THRESHOLD_");
+    if (thresh)
+      return atoll(thresh);
+    return (32 << 20);  // max for 64-bit Linux systems
```

Comment:
How about `128 KiB` ?

Reply:
We don't support 32-bit systems.

Reply:
I mean to use the default value of `M_MMAP_THRESHOLD` (according to the docs) if `MALLOC_MMAP_THRESHOLD_` is not set,

Reply:
M_MMAP_THRESHOLD by default is adaptive - it starts with 128k and then grows to the possible max as the user keeps freeing memory. In the end we would likely end up slowing the allocations down.